plugins/inputs/syslog/README.md

```diff
@@ -0,0 +1,110 @@
+# syslog input plugin
+
+Collects syslog messages as per RFC5425 or RFC5426.
+
+It can act as a syslog transport receiver over TLS (or TCP) - ie., RFC5425 - or over UDP - ie., RFC5426.
+
+This plugin listens for syslog messages following RFC5424 format. When received it parses them extracting metrics.
+
+### Configuration:
+
+#### TCP
+
+The minimal configuration is the following:
+
+```toml
+[[inputs.syslog]]
+  address = ":6514"
+```
+
+This starts this plugins as a syslog receiver over TCP protocol on port 6514.
+
+#### TLS
+
+To configure it as a TLS syslog receiver as recommended by RFC5425 give it the following configuration:
+
+```toml
+[[inputs.syslog]]
+  address = ":6514"
+  tls_cacert = "/etc/telegraf/ca.pem"
+  tls_cert = "/etc/telegraf/cert.pem"
+  tls_key = "/etc/telegraf/key.pem"
+```
+
+#### UDP
+
+To configure this plugin as per RFC5426 give it the following configuration:
+
+```toml
+[[inputs.syslog]]
+  protocol = "udp"
+  address = ":6514"
+```
+
+#### Other configs
+
+Other available configurations are:
+
+- `keep_alive_period`, `max_connections` for stream sockets
+- `best_effort` to tell the parser to work until it is able to do and extract partial but valid info
+
+### Metrics
+
+- syslog
+  - fields
+    - **version** (`uint16`)
+    - timestamp (`time.Time`)
+    - procid (`string`)
+    - msgid (`string`)
+    - _structureddata element id_ (`bool`)
+    - _structureddata element parameter name_ (`string`)
+  - tags
+    - **severity** (`string`)
+    - **severity_level** (`string`)
+    - **facility** (`string`)
+    - **facility_message** (`string`)
+    - hostname (`string`)
+    - appname (`string`)
+
+The name of fields in _italic_ corresponds to their runtime value.
+
+The fields/tags which name is in **bold** will always be present when a valid Syslog message has been received.
+
+### Syslog transport sender
```

Comment:
Title this section something like `RSYSLOG Integration`

Reply:
Ok.